pallets/author-mapping/src/lib.rs

```diff
@@ -56,15 +60,71 @@ pub mod pallet {
 		/// Currency in which the security deposit will be taken.
 		type DepositCurrency: Currency<Self::AccountId> + ReservableCurrency<Self::AccountId>;
 		/// The amount that should be taken as a security deposit when registering an AuthorId.
-		type DepositAmount: Get<<Self::DepositCurrency as Currency<Self::AccountId>>::Balance>;
+		type DepositAmount: Get<BalanceOf<Self>>;
 
 		/// A rough preliminary check to determine whether an account can make a new registration.
 		/// If you don't wish to do any such check, just return `true`.
 		fn can_register(account: &Self::AccountId) -> bool;
 	}
 
 	#[pallet::hooks]
-	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {}
+	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
+		fn on_runtime_upgrade() -> Weight {
+			//TODO we should consider moving the migrations to a dedicated module.
+			// Here's an example from the Substrate repo.
+			// github.com/paritytech/substrate/tree/master/frame/elections-phragmen/src/migrations
+			// I'm not sure where / when these migrations are invoked though.
+
+			// Get the runtime version out of storage
+			let maybe_storage_version = <Self as GetPalletVersion>::storage_version();
+			log::info!(
+				target: "author-mapping-migration",
+				"Running migration for pallet-author-mapping with storage version {:?}",
+				maybe_storage_version,
+			);
+
+			// This is the first migration we've ever done, and also the first time we're putting any
+			// version into storage (unless frame does something implicitly)
+			// So we only do the migration if there is no version already set. For future migrations
+			// we should check this version and ensure it is less than the upcoming version
+			if maybe_storage_version.is_some(){
+				// Exit early because this migration is already applied.
+				return 0u32.into()
+			}
```

Comment:
@thiolliere or @apopiak could you guys point me in the right direction here? I'm not clear how I'm supposed to use to use `PalletInfo`. I'm trying to write a fairly straight-forward migration that reserves more tokens for accounts who already had the old lower amount reserved.

Reply:
for me the easiest is to not write the migration in the pallet itself.
The pallet should provide a function to make the migration, in the runtime declaration this function should be called in the custom runtime upgrade (the generic OnCustomRuntimeUpgrade of executive type).
And once the runtime upgrade is done, you can remove it from the on custom runtime upgrade.

But some people prefer doing it in the pallet itself.

The version of the pallet is stored implicitly on each runtime upgrade (no matter what the pallet writes in its `on_runtime_upgrade` hook) since v3.0.0. So if the current runtime is using a substrate version v3.0.0 then the version is already stored.

I don't understand why you want to use PalletInfo though. PalletInfo contains the name and the index of the pallet

Reply:
> The version of the pallet is stored implicitly on each runtime upgrade

Can you tell me more about that? Where do I define it? Is it the semver from the cargo.toml?

> The pallet should provide a function to make the migration, in the runtime declaration this function should be called

Okay, that makes sense to me too. I'll try that out.

> I don't understand why you want to use PalletInfo

Yeah, I meant `PalletVersion` That was just a typo.

Reply:
> > The version of the pallet is stored implicitly on each runtime upgrade
> 
> Can you tell me more about that? Where do I define it? Is it the semver from the cargo.toml?

Yes the pallet version is the version found in cargo.toml
Since substrate 3.0.0 each runtime upgrade writes the version of pallet into storage.
(if you wonder about the implementation, the macro implements OnRuntimeUpgrade using the logic provided by user (or no logic if user doesn't sepcify anything) and add the storing of version at the end.)